### ISSUE TITLE
Wide Review Comment 2017: default font size and line heights

### ISSUE THREAD

**Issue by nigelmegitt** [w3c/webvtt#371]
Copy/paste from https://lists.w3.org/Archives/Public/public-tt/2017Sep/0080.html - raising as an issue for tracking/disposition purposes.

The default font size is 5vh, however the default line spacing for addressing line positions is only 5.33vh, representing a line spacing of 106.6% - that would represent an unusually small line height and may result in strange effects. I cannot see anywhere that overrides the line position height based on the computed value of the line-height property.

**Comment by silviapfeiffer** [comment]
vh stands for viewport height. The line height of 5.33vh stems from CEA-608/708 which provide for 15 lines of content on 80% height, so that gives each line 5.33% height.

NOTE: 5.33vh is only used for regions, not for cues, where line-height is simply the CSS default "normal", which is typically about 1.2% larger.

So we just need to fix this for regions.
I did some forensics on how this happened.

1/ Originally, for regions, line height was fixed to 5.33vh and font-size to 5.33/1.3vh giving sufficient space. https://github.com/w3c/webvtt/pull/242 removed that because of https://www.w3.org/Bugs/Public/show_bug.cgi?id=25636 .
2/ We also still have #269 .

Now, how to move forward?
I'd rather not change the 5vh default font-size on cues because it's already implemented everywhere and seems to work quite well.
I think we could just replace the 5.33vh with "normal" or with 6vh (which is 5*1.2) to provide more space. That means we ignore the inspiration from CEA608/708, which is fine - if somebody needs it fixed to what CEA608/708 do, they can manually set the font-size and line-height through CSS in STYLE, which is what they have to do already.

Would 6vh work for you?

**Comment by nigelmegitt** [comment]
OK, changing the default line height seems reasonable - I like the idea of clearly specified values so 6vh seems good, but I'm also aware that "normal" may give better results in terms of readability, so I don't have a strong position there.

Is there a place in the rendering algorithm where the line spacing for addressing line positions is obtained from the applicable style if set? I could not see any steps that adjusted _lineHeight_ from the initial 5.33vh value. Indeed it only appears in §6.1 subsection 12.1 bullet 2. So there appears to be no opportunity to change _lines_ based on an actual value.

**Comment by silviapfeiffer** [comment]
We have a fixed line height for regions so we're able to calculate the region's height. _lineHeight_ is adjusted when applying the terms of the CSS.

I'll set the line height for regions to 6vh.

We're tracking down how _lineHeight_ converts to the calculation of the region's height in #377 .

**Comment by silviapfeiffer** [comment]
@nigelmegitt PR for this issue is in https://github.com/w3c/webvtt/pull/411 - please review and comment whether you're happy that it addresses this issue?

**Comment by nigelmegitt** [comment]
> We have a fixed line height for regions so we're able to calculate the region's height.

Does that mean that if the user sets a different line height then the number of lines visible in the region is not what might be expected, i.e. the value of `lines`?

> @nigelmegitt PR for this issue is in #411 - please review and comment whether you're happy that it addresses this issue?

I've approved the PR as addressing the first part of this issue.

**Comment by silviapfeiffer** [comment]
> Does that mean that if the user sets a different line height then the number of lines visible in the region is not what might be expected, i.e. the value of lines?

@nigelmegitt no, the number of lines will react to the change in line height during applying the terms of the CSS.

**Comment by nigelmegitt** [comment]
> no, the number of lines will react to the change in line height during applying the terms of the CSS.

@silviapfeiffer sorry I wasn't clear - I get that the CSS specified line height is supposed to affect the height of each line in the region, but is the region height also affected by the CSS specified line height of the text within the region? It seems like no. So if the author sizes a region for 4 lines using the 6vh standard height of a line, but then applies a CSS line height that is 200% (i.e. 12vh) to the cue lines inside that region then only 2 lines would actually fit. Is that a correct understanding of the spec?

**Comment by silviapfeiffer** [comment]
@nigelmegitt the height of the region is specific in lines. So if the line height doubles, the region height doubles also.

**Comment by nigelmegitt** [comment]
@silviapfeiffer ok that's great, thank you. I just wasn't sure that is what the spec actually says. In the rendering section:

> Let _lines_ be lineHeight multiplied by regionHeight.

And then I couldn't find anywhere that modified this based on the line height of a cue line (I think it's supposed to be the first, but I'm not sure how "first" is defined).

**Comment by silviapfeiffer** [comment]
Adjustments are made in the next step:

> Apply the terms of the CSS specifications to regionNode

You might want to read up on how CSS calculation line heights here:
https://www.w3.org/TR/CSS22/visudet.html#line-height
https://www.w3.org/TR/CSS22/visudet.html#Computing_heights_and_margins
and
https://www.w3.org/TR/CSS22/visudet.html#inline-box-height

**Comment by dwsinger** [comment]
do we need a note in the spec to explain this?

**Comment by silviapfeiffer** [comment]
What would you have me say?

**Comment by dwsinger** [comment]
something like that the phrase "Apply the terms of the CSS specifications to regionNode" may/will result in changes to a number of variables, including lineHeight etc.

That's not obvious to the ordinary reader

**Comment by nigelmegitt** [comment]
I agree with @dwsinger here - we need to say it. The CSS line height calculation isn't relevant to https://github.com/w3c/webvtt/issues/371#issuecomment-346792789 which is to point out that any adjustment of line height later, for example when applying CSS, isn't then retrofitted to the _lines_ value to adjust the actual height of the region.

**Comment by tairt** [comment]
@silviapfeiffer Can a PR been prepared based on @dwsinger proposal? This seem to find agreement.

**Comment by silviapfeiffer** [comment]
Yes, of course. Sorry - I'm really time-poor. Any PRs that you can prepare would be very welcome also! Let's see if I can get this one done now...

**Comment by silviapfeiffer** [comment]
Please review #430
